src/Integration.Vsix/SonarLintTagger/CancellableJobRunner.cs

```diff
@@ -26,161 +26,180 @@
 using Microsoft.VisualStudio.Threading;
 using SonarLint.VisualStudio.Integration.Vsix.Resources;
 
-namespace SonarLint.VisualStudio.Integration.Vsix
+namespace SonarLint.VisualStudio.Integration.Vsix.CancellableJob
 {
-    /// <summary>
-    /// Runs a series of operations on a separate background thread.
-    /// Can be cancelled.
-    /// </summary>
-    internal sealed class CancellableJobRunner
+    public interface ICancellableJobRunner
```

Comment:
Introduce an interface so classes don't need to reference the implementing type directly.

Previously the associated types (e.g. the status enum) where nested inside the class. Since they are no longer inside the class, I added a more specific namespace (line 29 above) so the types don't pollute the `SonarLint.VisualStudio.Integration` namespace.